src/main/java/org/jenkinsci/plugins/durabletask/WindowsBatchScript.java

```diff
@@ -1,116 +1,149 @@
-    private boolean capturingOutput;
-
-    @DataBoundConstructor public WindowsBatchScript(String script) {
-        this.script = script;
-    }
-    
-    public String getScript() {
-        return script;
-    }
-
-    @Override public void captureOutput() {
-        capturingOutput = true;
-    }
-
-    @SuppressFBWarnings(value="VA_FORMAT_STRING_USES_NEWLINE", justification="%n from master might be \\n")
-    @Override protected FileMonitoringController doLaunch(FilePath ws, Launcher launcher, TaskListener listener, EnvVars envVars) throws IOException, InterruptedException {
-        if (launcher.isUnix()) {
-            throw new IOException("Batch scripts can only be run on Windows nodes");
-        }
-        BatchController c = new BatchController(ws);
-
-        String cmd;
-        if (capturingOutput) {
-            cmd = String.format("@echo off \r\ncmd /c \"\"%s\"\" > \"%s\" 2> \"%s\"\r\necho %%ERRORLEVEL%% > \"%s\"\r\n",
-                quote(c.getBatchFile2(ws)),
-                quote(c.getOutputFile(ws)),
-                quote(c.getLogFile(ws)),
-                quote(c.getResultFile(ws)));
-        } else {
-            cmd = String.format("@echo off \r\ncmd /c \"\"%s\"\" > \"%s\" 2>&1\r\necho %%ERRORLEVEL%% > \"%s\"\r\n",
-                quote(c.getBatchFile2(ws)),
-                quote(c.getLogFile(ws)),
-                quote(c.getResultFile(ws)));
-        }
-        c.getBatchFile1(ws).write(cmd, "UTF-8");
-        c.getBatchFile2(ws).write(script, "UTF-8");
-
-        Launcher.ProcStarter ps = launcher.launch().cmds("cmd", "/c", "\"\"" + c.getBatchFile1(ws) + "\"\"").envs(envVars).pwd(ws).quiet(true);
-        listener.getLogger().println("[" + ws.getRemote().replaceFirst("^.+\\\\", "") + "] Running batch script"); // details printed by cmd
-        /* Too noisy, and consumes a thread:
-        ps.stdout(listener);
-        */
-        ps.readStdout().readStderr(); // TODO see BourneShellScript
-        ps.start();
-        return c;
-    }
-
-    private static String quote(FilePath f) {
-        return f.getRemote().replace("%", "%%");
-    }
-
-    private static final class BatchController extends FileMonitoringController {
-        private BatchController(FilePath ws) throws IOException, InterruptedException {
-            super(ws);
-        }
-
-        public FilePath getBatchFile1(FilePath ws) throws IOException, InterruptedException {
-            return controlDir(ws).child("jenkins-wrap.bat");
-        }
-
-        public FilePath getBatchFile2(FilePath ws) throws IOException, InterruptedException {
-            return controlDir(ws).child("jenkins-main.bat");
-        }
-
-        private static final long serialVersionUID = 1L;
-    }
-
-    @Extension public static final class DescriptorImpl extends DurableTaskDescriptor {
-
-        @Override public String getDisplayName() {
-            return Messages.WindowsBatchScript_windows_batch();
-        }
-
-    }
-
-}
+/*
+ * The MIT License

```

Comment:
Please create minimal diffs. In particular do not introduce carriage returns into repository sources.